docs/examples/bifacial/plot_irradiance_nonuniformity_loss.py

```diff
@@ -0,0 +1,265 @@
+#    :doi:`10.5281/zenodo.11125039`.
+#
+# .. sectionauthor:: Echedey Luis <echelual (at) gmail.com>
+
+import numpy as np
+import pandas as pd
+import scipy.stats
+import matplotlib.pyplot as plt
+from matplotlib.cm import ScalarMappable
+from matplotlib.colors import Normalize
+
+from pvlib.pvsystem import nonuniform_irradiance_loss
+
+# %%
+# Theoretical and straightforward problem
+# ---------------------------------------
+# Let's set a fixed irradiance to each cell row of the PV array with the values
+# described in Figure 1 (a), [1]_. We will cover this case for educational
+# purposes.
+# Here we set and plot the global irradiance levels of each cell.
+
+x = np.arange(6, 0, -1)
+y = np.arange(6, 0, -1)
+cells_irrad = np.repeat([1059, 976, 967, 986, 1034, 1128], 6).reshape(6, 6)
+
+color_map = "gray"
+color_norm = Normalize(930, 1150)
+
+fig, ax = plt.subplots()
+fig.suptitle("Global Irradiance Levels of Each Cell")
+fig.colorbar(
+    ScalarMappable(cmap=color_map, norm=color_norm),
+    ax=ax,
+    orientation="vertical",
+    label="$[W/m^2]$",
+)
+ax.set_aspect("equal")
+ax.pcolormesh(
+    x,
+    y,
+    cells_irrad,
+    shading="nearest",
+    edgecolors="black",
+    cmap=color_map,
+    norm=color_norm,
+)
+
+# %%
+# Relative Mean Absolute Deviation
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# Calculate the Relative Mean Absolute Deviation (RMAD) of the cells'
+# irradiance with the following function, Eq. (4) of [1]_:
+#
+# .. math::
+#
+#    \Delta \left[ \% \right] = \frac{1}{n^2 \bar{G}_{total}}
+#    \sum_{i=1}^{n} \sum_{j=1}^{n} \lvert G_{total,i} - G_{total,j} \rvert
+#
+
+
+def rmad(data, axis=None):
+    """
+    Relative Mean Absolute Deviation.
+    https://stackoverflow.com/a/19472336/19371110
+    """
+    mad = np.mean(np.absolute(data - np.mean(data, axis)), axis)
+    return mad / np.mean(data, axis)
+
+
+rmad_cells = rmad(cells_irrad)
+
+# this is the same as a column's RMAD!
+print(rmad_cells == rmad(cells_irrad[:, 0]))
+
+# %%
+# Mismatch Loss
+# ^^^^^^^^^^^^^
+# Calculate the power loss percentage due to the irradiance non-uniformity
+# with the function :py:func:`pvlib.pvsystem.nonuniform_irradiance_loss`.
+
+mismatch_loss = nonuniform_irradiance_loss(rmad_cells)
+
+# %%
+# Total power incident on the module cells
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# It is the sum of the irradiance of each cell
+
+total_irrad = np.sum(cells_irrad)
+
+# %%
+# Mismatch-corrected irradiance
+# ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+# The power incident on the module cells is multiplied by the mismatch loss
+# as follows:
+
+mismatch_corrected_irrad = total_irrad * (1 - mismatch_loss)
+
+# %%
+# Results
+# ^^^^^^^
+
+print(f"Total power incident on the module cells: {total_irrad}")
+print(f"RMAD of the cells' irradiance: {rmad_cells}")
+print(f"Power loss % due to the irradiance non-uniformity: {mismatch_loss}")
+print(f"Effective power after mismatch correction: {mismatch_corrected_irrad}")
+
+# %%
+# A practical approach
```

Comment:
Is this section needed?  I worry it is a distraction from the main model.  Maybe it would be better to just mention that rear-side irradiance nonuniformity can be simulated using tools like solarfactors and bifacial_radiance?

Reply:
Mmmm I share the same view. @cesardd gave me a lot of feedback regarding this model yesterday and so I'm going to do a good amount of changes to the example. Some points are:

In the theoretical approach:

* Simplify unneeded calculus and results
* Make the plot more similar to the paper example

Regarding the practical example, I'd like to keep it as a showcase of the RMAD properties. I'm afraid some user may use it directly with the backside RMAD. Also, I think this content also has some value from the didactic point of view.

I plan future changes in this section to remove the polynomial models and use real data to show and plot the modelled vs measured mismatch with data provided from my thesis tutors.

Anyway, tell me if you prefer to nix this part. In any case I will mention those tools for the backside irrad RMAD calculation in the docstring.